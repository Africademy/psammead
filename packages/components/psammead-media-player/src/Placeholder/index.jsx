import React from 'react';
import styled from 'styled-components';
import { string, func, shape, oneOf } from 'prop-types';
import Image from '@bbc/psammead-image';
import PlayButton from '@bbc/psammead-play-button';
import { getSansRegular } from '@bbc/psammead-styles/font-styles';
import { GEL_SPACING_DBL, GEL_SPACING } from '@bbc/gel-foundations/spacings';
import { GEL_GROUP_2_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';
import { GEL_LONG_PRIMER } from '@bbc/gel-foundations/typography';
import { C_POSTBOX, C_WHITE } from '@bbc/psammead-styles/colours';
import Guidance from '../Guidance';

const StyledPlaceholder = styled.div`
  cursor: pointer;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`;

const StyledNoScript = styled.noscript`
  ${({ service }) => getSansRegular(service)}
  ${GEL_LONG_PRIMER};

  position: absolute;
  bottom: 0;
  font-weight: normal;
  padding: ${GEL_SPACING};
  color: ${C_WHITE};
  border-bottom: 0.0625rem solid transparent;
  @media screen and (-ms-high-contrast: active) {
    background-color: window;
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: ${GEL_SPACING_DBL};
  }
`;

const PlayButtonWrapper = styled.span``;

const StyledPlayButton = styled(PlayButton)`
  position: absolute;
  bottom: 0;

  /* stylelint-disable */
  /* https://www.styled-components.com/docs/advanced#referring-to-other-components */
  ${StyledPlaceholder}:hover &,
  ${StyledPlaceholder}:focus & {
    background-color: ${C_POSTBOX};
  }
  /* stylelint-enable */
`;

const NoScriptHiddenStyle = () => (
  <noscript>
    <style>
      {`
     #media-player-guiidance, #media-player-play-button {
       display: none;
     }
    `}
    </style>
  </noscript>
);

const Placeholder = ({ onClick, service, src, srcset, mediaInfo }) => {
  const {
    title,
    datetime,
    duration,
    durationSpoken,
    type,
    guidanceMessage,
  } = mediaInfo;

  return (
    <StyledPlaceholder onClick={onClick}>
      <NoScriptHiddenStyle />
      {guidanceMessage && (
        <Guidance
          messageId="media-player-guiidance"
          service={service}
          guidanceMessage={guidanceMessage}
        />
      )}
      <PlayButtonWrapper id="media-player-play-button">
        <StyledPlayButton
          title={title}
          service={service}
          onClick={() => {}}
          datetime={datetime}
          duration={duration}
          durationSpoken={durationSpoken}
          type={type}
          guidanceMessage={guidanceMessage}
        />
      </PlayButtonWrapper>

      <StyledNoScript service={service}>
        This video cannot play in your browser. Please enable Javascript or try
        a different browser.
      </StyledNoScript>
      <Image alt="" src={src} srcset={srcset} />
    </StyledPlaceholder>
  );
};

Placeholder.propTypes = {
  onClick: func.isRequired,
  service: string.isRequired,
  src: string.isRequired,
  srcset: string,
  mediaInfo: shape({
    title: string.isRequired,
    datetime: string,
    duration: string,
    durationSpoken: string,
    type: oneOf(['audio', 'video']),
    guidanceMessage: string,
  }),
};
Placeholder.defaultProps = {
  srcset: null,
  mediaInfo: shape({
    datetime: null,
    duration: null,
    durationSpoken: null,
    type: 'video',
    guidanceMessage: null,
  }),
};

export default Placeholder;
