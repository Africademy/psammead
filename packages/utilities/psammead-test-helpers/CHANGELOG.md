# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.1.2   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
