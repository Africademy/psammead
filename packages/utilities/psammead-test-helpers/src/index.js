import { render, waitForDomChange } from '@testing-library/react';
import 'jest-styled-components';
import deepClone from 'ramda/src/clone';

const renderWithHelmet = async component => {
  /*
 * Similar to this problem https://github.com/testing-library/react-testing-library/issues/402
 * This helper was created to solve the problem of rendering helmet/head contents in snapshots.
 * Pass in a component that uses helmet somewhere in the component tree.
 * The full html tree is rendered and returned which you can then use to snapshot helmet/head contents.
 * Example of use in a test file:
 it('should render correctly', async () => {
    const html = await renderHelmet(<SomeComponent {...someProps} />);
    expect(html).toMatchSnapshot();
  });
 */

  const { container } = render(component);
  const noop = () => {};
  const ARBITRARY_TIMEOUT = 100; // seems long enough for any dom mutations to occur

  return waitForDomChange({
    container: document.querySelector('head'),
    timeout: ARBITRARY_TIMEOUT,
  })
    .catch(noop) // handle a waitForDomChange timeout
    .then(mutationsList => {
      const domMutationDetected = mutationsList && mutationsList.length;

      if (domMutationDetected) {
        // helmet was probably used so we should get the full html

        const htmlElement = document.querySelector('html');

        const helmetElements = document.querySelectorAll(
          '[data-react-helmet="true"]',
        );

        const removeHelmetAttributes = el =>
          el.removeAttribute('data-react-helmet'); // remove react-helmet attribute noise from elements

        removeHelmetAttributes(htmlElement); // remove react-helmet attribute noise from elements

        Array.from(helmetElements).forEach(removeHelmetAttributes);

        return { container: document.querySelector('html') };
      }

      return { container };
    });
};

export const shouldMatchSnapshot = (title, component) => {
  it(title, async () => {
    // select the first child to remove the pointless wrapping div from snapshots
    const removeWrappingDiv = container => container.firstChild;
    const { container } = await renderWithHelmet(component);
    const hasOneChild = container.children.length === 1;
    /*
     * if the container has more than one child then it's a component that uses a
     * fragment at the top level so we should not select the first child because it
     * wouldn't snapshot the whole component
     */
    expect(
      hasOneChild ? removeWrappingDiv(container) : container,
    ).toMatchSnapshot();
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const { container } = render(component);
    expect(container.firstChild).toBeNull();
  });
};

export const setWindowValue = (key, value) => {
  const windowValue = window[key];
  delete window[key];

  let newValue = value;

  if (value && typeof value === 'object') {
    newValue = {
      ...deepClone(windowValue),
      ...value,
    };
  }

  Object.defineProperty(window, key, {
    value: newValue,
    writable: true,
  });
};

export const resetWindowValue = (key, value) => {
  Object.defineProperty(window, key, {
    value,
    writable: true,
  });
};

export const suppressPropWarnings = warnings => {
  const { expectedWarnings } = window;
  if (expectedWarnings && Array.isArray(expectedWarnings)) {
    window.expectedWarnings = [...expectedWarnings, warnings];
  } else {
    window.expectedWarnings = [warnings];
  }
};

const errorIfMissingKey = (keys, object, message) => {
  keys.forEach(key => {
    if (!(key in object)) {
      throw new Error(`Missing value '${key}' in ${message}.`);
    }
  });
};

const checkKeysExistInBothObjects = (object1, object2, message1, message2) => {
  const object1Keys = Object.keys(object1);
  const object2Keys = Object.keys(object2);

  errorIfMissingKey(object1Keys, object2, message2);
  errorIfMissingKey(object2Keys, object1, message1);
};

const checkTypesOfExports = (
  actualExportsByName,
  actualExports,
  expectedExports,
  utilityName,
) => {
  actualExportsByName.forEach(actualExportName => {
    const actualExportValue = actualExports[utilityName][actualExportName];
    const expectedExport = expectedExports[utilityName][actualExportName];
    const typeCheck = typeof actualExportValue === expectedExport; // eslint-disable-line valid-typeof

    // if this fails it is likely that an export is missing from the unit test expectation
    expect(typeCheck).toBe(true);
  });
};

export const testUtilityPackages = (
  actualExports,
  expectedExports,
  packageName,
) => {
  const actualUtilities = Object.keys(actualExports);

  // check if the actual exported file has defined expected values to match and that the expected values are actually exported
  checkKeysExistInBothObjects(
    actualExports,
    expectedExports,
    `the actual utilities for '${packageName}'`,
    `the expected utilities for '${packageName}'`,
  );

  actualUtilities.forEach(utilityName => {
    const actualExportsByName = Object.keys(actualExports[utilityName]);

    // check if each of the actual exported consts have an expected value to match and that the expected values are actually exported
    checkKeysExistInBothObjects(
      actualExports[utilityName],
      expectedExports[utilityName],
      `the actual export for '${packageName}/${utilityName}'`,
      `the expected export for '${packageName}/${utilityName}'`,
    );

    checkTypesOfExports(
      actualExportsByName,
      actualExports,
      expectedExports,
      utilityName,
    );
  });
};
