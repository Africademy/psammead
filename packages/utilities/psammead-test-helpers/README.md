# psammead-test-helpers &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/BBC-News/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-test-helpers.svg)](https://www.npmjs.com/package/@bbc/psammead-test-helpers) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://reactjs.org/docs/how-to-contribute.html#your-first-pull-request)

This package is a collection helper methods used for snapshot testing, with support for styled components.

## Installation

```jsx
npm install @bbc/psammead-test-helpers --save
```

## Usage

```jsx
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';

shouldMatchSnapshot(
  'should render correctly',
  <h1>Hello World</h1>,
);
```

## Roadmap

There is currently a plan to migrate from `react-test-renderer` to `enzyme` as it provides a nicer higher-level API over the same functionality, as well as having support for more granular unit tests.

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/BBC-News/psammead/blob/latest/LICENSE).
