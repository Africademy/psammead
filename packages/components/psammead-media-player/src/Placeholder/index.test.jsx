import React from 'react';
import { render, fireEvent, getByText } from '@testing-library/react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import Placeholder from '.';

describe('Media Player: Placeholder', () => {
  const mockOnClick = jest.fn();
  const withDuration = {
    duration: '2:30',
    durationSpoken: '2 minutes 30 seconds',
    datetime: 'PT2M30S',
  };

  afterEach(() => {
    jest.resetModules();
  });

  shouldMatchSnapshot(
    'should render a video placeholder',
    <Placeholder
      onClick={mockOnClick}
      src="http://foo.bar/placeholder.png"
      service="news"
      mediaInfo={{ title: 'Dog chases cat.', ...withDuration }}
      className="foo"
    />,
  );

  shouldMatchSnapshot(
    'should render a video placeholder without duration',
    <Placeholder
      onClick={mockOnClick}
      src="http://foo.bar/placeholder.png"
      service="news"
      mediaInfo={{ title: 'Dog chases cat.', ...withDuration }}
      className="foo"
    />,
  );

  shouldMatchSnapshot(
    'should render an audio placeholder',
    <Placeholder
      onClick={mockOnClick}
      src="http://foo.bar/placeholder.png"
      service="news"
      mediaInfo={{ type: 'audio', title: 'Dog barks at cat.', ...withDuration }}
      className="foo"
    />,
  );

  shouldMatchSnapshot(
    'should render an audio placeholder without duration',
    <Placeholder
      onClick={mockOnClick}
      src="http://foo.bar/placeholder.png"
      service="news"
      mediaInfo={{ type: 'audio', title: 'Dog barks at cat.', ...withDuration }}
      className="foo"
    />,
  );

  it('should call onClick when the placeholder or play button is clicked', () => {
    const { container } = render(
      <Placeholder
        onClick={mockOnClick}
        src="http://foo.bar/placeholder.png"
        service="news"
        mediaInfo={{ title: 'Dog chases cat.', ...withDuration }}
        className="foo"
      />,
    );
    fireEvent.click(getByText(container.firstChild, '2:30'));
    fireEvent.click(container.firstChild);
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
