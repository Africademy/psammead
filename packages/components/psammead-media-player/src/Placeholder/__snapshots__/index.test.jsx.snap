// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Player: Placeholder should render a video placeholder 1`] = `
.emotion-10 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
}

@media (min-width:37.5rem) {
  .emotion-2 {
    font-size: 0.875rem;
  }
}

.emotion-0 {
  position: absolute;
  bottom: 0;
  display: none;
}

.emotion-6 {
  position: absolute;
  bottom: 0;
}

.emotion-11:hover .emotion-6,
.emotion-11:focus .emotion-6 {
  background-color: #B80000;
}

.emotion-4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-8 {
  display: block;
  width: 100%;
  visibility: visible;
}

<div>
  <div
    class="emotion-10 emotion-11"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <noscript
        class="emotion-0 emotion-1"
      />
    </div>
    <button
      class="Button-sc-34e3rb-0 kwlSXK emotion-6 emotion-7"
    >
      <span
        class="emotion-4 emotion-5"
      >
        Play video, "Dog chases cat.", 2 minutes 30 seconds
      </span>
      <div
        aria-hidden="true"
        class="IconWrapper-sc-34e3rb-1 jGavfg"
      >
        <svg
          aria-hidden="true"
          class="MediaIcon-sc-10lpekz-0 VideoMediaIcon-sc-10lpekz-1 gUVBrE"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </div>
      <time
        aria-hidden="true"
        class="TimeDuration-sc-34e3rb-2 cZLyLe"
        datetime="PT2M30S"
      >
        2:30
      </time>
    </button>
    <img
      alt=""
      class="emotion-8 emotion-9"
      src="http://foo.bar/placeholder.png"
    />
  </div>
</div>
`;

exports[`Media Player: Placeholder should render a video placeholder with guidance 1`] = `
.emotion-12 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-2 {
  position: absolute;
  bottom: 0;
  display: none;
}

.emotion-8 {
  position: absolute;
  bottom: 0;
}

.emotion-13:hover .emotion-8,
.emotion-13:focus .emotion-8 {
  background-color: #B80000;
}

.emotion-6 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-10 {
  display: block;
  width: 100%;
  visibility: visible;
}

.emotion-4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
  background-color: rgba(34,34,34,0.75);
}

@media (min-width:37.5rem) {
  .emotion-4 {
    font-size: 0.875rem;
  }
}

@media screen and (-ms-high-contrast:active) {
  .emotion-4 {
    background-color: transparent;
  }
}

.emotion-0 {
  display: block;
  font-weight: normal;
  padding: 0.5rem;
  border-bottom: 0.0625rem solid transparent;
}

@media screen and (-ms-high-contrast:active) {
  .emotion-0 {
    background-color: window;
  }
}

@media (min-width:25rem) {
  .emotion-0 {
    padding: 1rem;
  }
}

<div>
  <div
    class="emotion-12 emotion-13"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <strong
        aria-hidden="true"
        class="guidance-message emotion-0 emotion-1"
      >
        Guidance: May contain strong language, sexual or violent content that may offend.
      </strong>
      <noscript
        class="emotion-2 emotion-3"
      />
    </div>
    <button
      class="Button-sc-34e3rb-0 kwlSXK emotion-8 emotion-9"
    >
      <span
        class="emotion-6 emotion-7"
      >
        Guidance: May contain strong language, sexual or violent content that may offend. Play video, "Dog chases cat.", 2 minutes 30 seconds
      </span>
      <div
        aria-hidden="true"
        class="IconWrapper-sc-34e3rb-1 jGavfg"
      >
        <svg
          aria-hidden="true"
          class="MediaIcon-sc-10lpekz-0 VideoMediaIcon-sc-10lpekz-1 gUVBrE"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </div>
      <time
        aria-hidden="true"
        class="TimeDuration-sc-34e3rb-2 cZLyLe"
        datetime="PT2M30S"
      >
        2:30
      </time>
    </button>
    <img
      alt=""
      class="emotion-10 emotion-11"
      src="http://foo.bar/placeholder.png"
    />
  </div>
</div>
`;

exports[`Media Player: Placeholder should render a video placeholder without duration 1`] = `
.emotion-10 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
}

@media (min-width:37.5rem) {
  .emotion-2 {
    font-size: 0.875rem;
  }
}

.emotion-0 {
  position: absolute;
  bottom: 0;
  display: none;
}

.emotion-6 {
  position: absolute;
  bottom: 0;
}

.emotion-11:hover .emotion-6,
.emotion-11:focus .emotion-6 {
  background-color: #B80000;
}

.emotion-4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-8 {
  display: block;
  width: 100%;
  visibility: visible;
}

<div>
  <div
    class="emotion-10 emotion-11"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <noscript
        class="emotion-0 emotion-1"
      />
    </div>
    <button
      class="Button-sc-34e3rb-0 kwlSXK emotion-6 emotion-7"
    >
      <span
        class="emotion-4 emotion-5"
      >
        Play video, "Dog chases cat."
      </span>
      <div
        aria-hidden="true"
        class="IconWrapper-sc-34e3rb-1 bShuUP"
      >
        <svg
          aria-hidden="true"
          class="MediaIcon-sc-10lpekz-0 VideoMediaIcon-sc-10lpekz-1 gUVBrE"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </div>
    </button>
    <img
      alt=""
      class="emotion-8 emotion-9"
      src="http://foo.bar/placeholder.png"
    />
  </div>
</div>
`;

exports[`Media Player: Placeholder should render an audio placeholder 1`] = `
.emotion-10 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
}

@media (min-width:37.5rem) {
  .emotion-2 {
    font-size: 0.875rem;
  }
}

.emotion-0 {
  position: absolute;
  bottom: 0;
  display: none;
}

.emotion-6 {
  position: absolute;
  bottom: 0;
}

.emotion-11:hover .emotion-6,
.emotion-11:focus .emotion-6 {
  background-color: #B80000;
}

.emotion-4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-8 {
  display: block;
  width: 100%;
  visibility: visible;
}

<div>
  <div
    class="emotion-10 emotion-11"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <noscript
        class="emotion-0 emotion-1"
      />
    </div>
    <button
      class="Button-sc-34e3rb-0 kwlSXK emotion-6 emotion-7"
    >
      <span
        class="emotion-4 emotion-5"
      >
        Play audio, "Dog barks at cat.", 2 minutes 30 seconds
      </span>
      <div
        aria-hidden="true"
        class="IconWrapper-sc-34e3rb-1 jGavfg"
      >
        <svg
          aria-hidden="true"
          class="MediaIcon-sc-10lpekz-0 AudioMediaIcon-sc-10lpekz-2 jqcLqP"
          focusable="false"
          height="12px"
          viewBox="0 0 13 12"
          width="13px"
        >
          <path
            d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
          />
          <path
            d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
          />
        </svg>
      </div>
      <time
        aria-hidden="true"
        class="TimeDuration-sc-34e3rb-2 cZLyLe"
        datetime="PT2M30S"
      >
        2:30
      </time>
    </button>
    <img
      alt=""
      class="emotion-8 emotion-9"
      src="http://foo.bar/placeholder.png"
    />
  </div>
</div>
`;

exports[`Media Player: Placeholder should render an audio placeholder without duration 1`] = `
.emotion-10 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
}

@media (min-width:37.5rem) {
  .emotion-2 {
    font-size: 0.875rem;
  }
}

.emotion-0 {
  position: absolute;
  bottom: 0;
  display: none;
}

.emotion-6 {
  position: absolute;
  bottom: 0;
}

.emotion-11:hover .emotion-6,
.emotion-11:focus .emotion-6 {
  background-color: #B80000;
}

.emotion-4 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-8 {
  display: block;
  width: 100%;
  visibility: visible;
}

<div>
  <div
    class="emotion-10 emotion-11"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <noscript
        class="emotion-0 emotion-1"
      />
    </div>
    <button
      class="Button-sc-34e3rb-0 kwlSXK emotion-6 emotion-7"
    >
      <span
        class="emotion-4 emotion-5"
      >
        Play audio, "Dog barks at cat."
      </span>
      <div
        aria-hidden="true"
        class="IconWrapper-sc-34e3rb-1 bShuUP"
      >
        <svg
          aria-hidden="true"
          class="MediaIcon-sc-10lpekz-0 AudioMediaIcon-sc-10lpekz-2 jqcLqP"
          focusable="false"
          height="12px"
          viewBox="0 0 13 12"
          width="13px"
        >
          <path
            d="M9.021 1.811l-.525.525c.938.938 1.5 2.25 1.5 3.675s-.563 2.738-1.5 3.675l.525.525c1.05-1.087 1.725-2.55 1.725-4.2s-.675-3.112-1.725-4.2z"
          />
          <path
            d="M10.596.199l-.525.562c1.35 1.35 2.175 3.225 2.175 5.25s-.825 3.9-2.175 5.25l.525.525c1.5-1.462 2.4-3.525 2.4-5.775s-.9-4.312-2.4-5.812zM6.996 1.511l-2.25 2.25H.996v4.5h3.75l2.25 2.25z"
          />
        </svg>
      </div>
    </button>
    <img
      alt=""
      class="emotion-8 emotion-9"
      src="http://foo.bar/placeholder.png"
    />
  </div>
</div>
`;

exports[`Media Player: Placeholder should render no-js styles when noJsClassName prop is used 1`] = `
.emotion-6 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-10 {
  display: block;
  width: 100%;
  visibility: visible;
}

.emotion-0 {
  display: block;
  font-weight: normal;
  padding: 0.5rem;
  border-bottom: 0.0625rem solid transparent;
}

@media screen and (-ms-high-contrast:active) {
  .emotion-0 {
    background-color: window;
  }
}

@media (min-width:25rem) {
  .emotion-0 {
    padding: 1rem;
  }
}

.emotion-12 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: default;
}

.emotion-4 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
  background-color: rgba(34,34,34,0.75);
}

@media (min-width:37.5rem) {
  .emotion-4 {
    font-size: 0.875rem;
  }
}

@media screen and (-ms-high-contrast:active) {
  .emotion-4 {
    background-color: transparent;
  }
}

.no-js .emotion-4 {
  background-color: rgba(34,34,34,0.75);
}

@media screen and (-ms-high-contrast:active) {
  .no-js .emotion-4 {
    background-color: transparent;
  }
}

.no-js .emotion-4 .guidance-message {
  display: none;
}

.emotion-2 {
  position: absolute;
  bottom: 0;
}

.emotion-8 {
  position: absolute;
  bottom: 0;
}

.no-js .emotion-8 {
  display: none;
}

.emotion-13:hover .emotion-8,
.emotion-13:focus .emotion-8 {
  background-color: #B80000;
}

<div>
  <div
    class="emotion-12 emotion-13"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <strong
        aria-hidden="true"
        class="guidance-message emotion-0 emotion-1"
      >
        Guidance: May contain strong language, sexual or violent content that may offend.
      </strong>
      <noscript
        class="emotion-2 emotion-3"
      />
    </div>
    <button
      class="Button-sc-34e3rb-0 kwlSXK emotion-8 emotion-9"
    >
      <span
        class="emotion-6 emotion-7"
      >
        Guidance: May contain strong language, sexual or violent content that may offend. Play video, "Dog chases cat.", 2 minutes 30 seconds
      </span>
      <div
        aria-hidden="true"
        class="IconWrapper-sc-34e3rb-1 jGavfg"
      >
        <svg
          aria-hidden="true"
          class="MediaIcon-sc-10lpekz-0 VideoMediaIcon-sc-10lpekz-1 gUVBrE"
          focusable="false"
          height="12px"
          viewBox="0 0 32 32"
          width="12px"
        >
          <polygon
            points="3,32 29,16 3,0"
          />
        </svg>
      </div>
      <time
        aria-hidden="true"
        class="TimeDuration-sc-34e3rb-2 cZLyLe"
        datetime="PT2M30S"
      >
        2:30
      </time>
    </button>
    <img
      alt=""
      class="emotion-10 emotion-11"
      src="http://foo.bar/placeholder.png"
    />
  </div>
</div>
`;
